src/library_pthread.js

```diff
@@ -207,6 +207,10 @@ var LibraryPThread = {
     },
 
     threadCancel: function() {
+      // Disable all cancellation so that executing the cleanup handlers won't trigger another JS
+      // canceled exception to be thrown.
+      Atomics.store(HEAPU32, (threadInfoStruct + {{{ C_STRUCTS.pthread.canceldisable }}} ) >> 2, 1/*PTHREAD_CANCEL_DISABLE*/);
+      Atomics.store(HEAPU32, (threadInfoStruct + {{{ C_STRUCTS.pthread.cancelasync }}} ) >> 2, 0/*PTHREAD_CANCEL_DEFERRED*/);
       PThread.runExitHandlers();
       Atomics.store(HEAPU32, (threadInfoStruct + {{{ C_STRUCTS.pthread.threadExitCode }}} ) >> 2, -1/*PTHREAD_CANCELED*/);
       Atomics.store(HEAPU32, (threadInfoStruct + {{{ C_STRUCTS.pthread.threadStatus }}} ) >> 2, 1); // Mark the thread as no longer running.
```

Comment:
These five lines are now a somewhat direct copy-paste of the above block. Perhaps refactor this commonality to its own `writeThreadExit(exitCode)` function that both call sites can call? (This block below would call it with exitCode==-1)

Reply:
@juj thanks for pointing that out. Done in acd89cd3e.